rpcserver.go

```diff
@@ -1302,6 +1306,48 @@ func (r *rpcServer) ListPeers(ctx context.Context,
 	return resp, nil
 }
 
+// ChangePassword changes the password of the currently loaded encrypted wallet,
+// so it must be unlocked first.
+func (r *rpcServer) ChangePassword(ctx context.Context,
+	in *lnrpc.ChangePasswordRequest) (*lnrpc.ChangePasswordResponse, error) {
+
+	publicPw := in.CurrentPassword
+	privatePw := in.CurrentPassword
+
+	// If the current password is blank, we'll assume the user is coming
+	// from a --noencryptwallet state, so we'll use the default passwords.
+	if len(in.CurrentPassword) == 0 {
+		publicPw = defaultPublicWalletPw
+		privatePw = defaultPrivateWalletPw
+	}
+
+	// Require the new password to have a minimum length of 8 characters.
+	if len(in.NewPassword) < 8 {
+		return nil, fmt.Errorf("password must have at least 8 " +
+			"characters")
+	}
+
+	// Attempt to change both the public and private passphrases for the
+	// wallet. This will be done atomically in order to prevent one
+	// passphrase change from being successful and not the other.
+	err := r.server.cc.wallet.ChangePassphrases(
+		publicPw, in.NewPassword, privatePw, in.NewPassword,
+	)
+	if err != nil {
+		return nil, fmt.Errorf("unable to change password: %v", err)
+	}
+
+	// We'll also need to update the encryption key to our macaroons
```

Comment:
What happens here if we change the password, but then crash? In this case, the macaroons would still be encrypted under the prior password. One way to resolve this would be to mark the macaroon encryption state as "pending change". Upon restart, if we detect in this state, then we can try to decrypt under the new passphrase, and if not, we'd rotate over to the new passphrase.

Reply:
I like this idea! This helped me come up with a potential solution:

Rather than changing the macaroon encryption key, we can create an empty file `.regenerate_macaroons` (or something similar) within the `lnddir` and check if it exists while the daemon is starting. If it does, we can delete the previous `macaroon.db` and `*.macaroon` files, which is safe, and recreate the new `macaroon.db` and `*.macaroon` files using the new password.

This would also allow us to move the `ChangePassword` RPC into the `WalletUnlocker` service easily.

Thoughts?

Reply:
The above can be simplified to deleting the macaroons whenever a password change happens. We can then move the `ChangePassword` RPC into the `WalletUnlocker` and generate the macaroon database and files with the new password on startup.

Reply:
Fixed.